### ISSUE TITLE
Errors when serializing / unserializing S4 objects twice

### ISSUE THREAD

**Issue by MatthiasHinz** [jeroen/jsonlite#174]
Hi,
I was looking for a way to convert between S4 objects and JSON and noticed that this could be done with
`serializeJSON / unserializeJSON `. This worked fine at force glance, but then I tried to convert a previously unserialized object again and experienced errors. 

Any ideas / solutions for that?

I made two minimal examples to illustrate that:

``` R
> library(jsonlite)
> setClass("myClass", slots = list(name = "character"))
> obj <- new("myClass", name = "myName")
> obj <- jsonlite::unserializeJSON(jsonlite::serializeJSON(obj))
> obj <- jsonlite::unserializeJSON(jsonlite::serializeJSON(obj))
Error: lexical error: invalid character inside string.
          ol","attributes":{},"value":["NULL"]},"access":{"type":"li
                     (right here) ------^
In addition: Warning messages:
1: Tried to use as.scalar on an object of length 0 
2: Tried to use as.scalar on an object of length 0 
> 
> 
> library(sp)
> demo(meuse, ask = FALSE, echo = FALSE)
> class(meuse)
[1] "SpatialPointsDataFrame"
attr(,"package")
[1] "sp"
> obj <- meuse
> obj <- jsonlite::unserializeJSON(jsonlite::serializeJSON(obj))
> obj <- jsonlite::unserializeJSON(jsonlite::serializeJSON(obj))
Error in .getClassFromCache(Class, where, package = package, resolve.confl = "none") : 
  class should be either a character-string name or a class definition
In addition: Warning messages:
1: Tried to use as.scalar on an object of length 0 
2: Tried to use as.scalar on an object of length 0 
3: Tried to use as.scalar on an object of length 0 
4: Tried to use as.scalar on an object of length 0 
```

**Comment by MatthiasHinz** [comment]
FYI

```
> sessionInfo()
R version 3.3.3 (2017-03-06)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 8 (jessie)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sp_1.2-4     jsonlite_1.3

loaded via a namespace (and not attached):
[1] tools_3.3.3     grid_3.3.3      lattice_0.20-34
```

**Comment by jeroen** [comment]
Thanks! This is now fixed in version 1.4: https://github.com/jeroen/jsonlite/commit/f7d0ea620b11072ef5259468a69fc01b9906c5e1
